### PR TITLE
Staff can delete their own FileSets.

### LINKED ISSUE #1367: Staff users can't delete FileSets they have created
The `Ability` class is checking whether the FileSet's `depositor` matches the user's id, but the FileSet's `depositor` property is `nil`.

### LINKED ISSUE #4379: `ability_spec.rb` tests its functionality without persisted resources, requiring code that never runs in production.
https://github.com/pulibrary/figgy/blob/dd172ae6404d12dee256c11112be286884bc2e4b/app/resources/ephemera_folders/ephemera_folder_decorator.rb#L59-L63

captures an ArgumentError solely because of

https://github.com/pulibrary/figgy/blob/dd172ae6404d12dee256c11112be286884bc2e4b/spec/models/ability_spec.rb#L624 and similar calls using `FactoryBot.create` instead of `FactoryBot.create_for_repository`, the former of which doesn't actually persist the resource.

### PR DESCRIPTION
**Pull request by tpendragon**
This also fixes up `ability.rb` so it uses the correct
`.create_for_repository`.

Closes #1367
Closes #4379

### CODE REVIEW

**Review by hackartisan — COMMENTED** [review]
👍🏻 but not sure what is going on with CI

**Comment by tpendragon** [comment]
Coverage should be fixed.

File: app/resources/ephemera_folders/ephemera_folder_decorator.rb [hunk 1]
```diff
@@ -58,8 +58,6 @@ def collections
   # TODO: Rename to decorated_ephemera_box
   def ephemera_box
     wayfinder.decorated_ephemera_box
-  rescue ArgumentError # Necessary because ability_spec.rb doesn't always save objects. TODO: Fix.
```
Comment (hackartisan) [inline app/resources/ephemera_folders/ephemera_folder_decorator.rb:61]: I don't really understand from this comment. Maybe you could write it in an issue?

**Review by hackartisan — CHANGES_REQUESTED** [review]

File: app/resources/ephemera_folders/ephemera_folder_decorator.rb (hunk 1 above)
Comment (tpendragon) [reply app/resources/ephemera_folders/ephemera_folder_decorator.rb:61]: Sure. See #4379.

Comment (hackartisan) [reply app/resources/ephemera_folders/ephemera_folder_decorator.rb:61]: Thank you!

**Review by hackartisan — APPROVED** [review]

**Pull request merged**